src/beyondllm/generator/generate.py

```diff
@@ -152,6 +152,27 @@ def get_ground_truth(self, answer:str, llm = None):
         
         return f"Ground truth score: {round(score, 1)}\n{thresholdCheck(score)}"
         
+    def get_faithfulness(self, llm=None):
+        if llm is None:
+            llm = self.llm
+        try:
+            score_str = llm.predict(FAITHFULNESS.format(ground_truth="", generated_response=self.RESPONSE))
+            score = float(extract_number(score_str))
+
+            return f"Faithfulness score: {round(score, 1)}\n{thresholdCheck(score)}"
+        except Exception as e:
+            return f"Failed during faithfulness evaluation: {e}"
+
+    def get_correctness(self, llm=None):
```

Comment:
You have to pass in the ground truth in the parameter, kindly add that in the parameter so that the user passes the ground truth when calling the function. Something like: 

```py
def get_correctness(self, answer:str, llm=None):
```

and then pass it in the ground truth:


```py
llm.predict(CORRECTNESS.format(ground_truth=answer, generated_response=self.RESPONSE))
```

maybe add a check as well to throw an error if the user does not provide the ground truth.

Reply:
Added the input param